src/Components/ContextProvider.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { RefObject } from 'react';
+import { Todo } from '../types/Todo';
+
+interface ContextProps {
+  todos: Todo[];
+  setTodos: (value: Todo[]) => void;
+  setErrorMessage: (value: string) => void;
+  inputRef: RefObject<HTMLInputElement>;
+}
+
+export const Context = React.createContext<ContextProps>({
+  todos: [],
+  setTodos: () => {},
+  setErrorMessage: () => {},
+  inputRef: { current: null },
+});
+
+export const ContextProvider = ({
+  children,
+  context,
+}: {
+  children: React.ReactNode;
+  context: ContextProps;
+}) => {
```

Comment:
```suggestion
export const TodosContext = React.createContext<TodosContextType>({
  todos: [],
  setTodos: () => {},
  filterStatus: TodoStatus.all,
  setFilterStatus: () => {},
  clearAllCompletedTodos: () => {},
  filteredTodos: [],
  inputFocus: { current: null },
});

type Props = {
  children: React.ReactNode;
};

export const TodosProvider: React.FC<Props> = ({ children }) => {
  const [todos, setTodos] = useState<Todo[]>(() => {
    const savedTodos = localStorage.getItem('todos');

    return savedTodos ? JSON.parse(savedTodos) : [];
  });

  const [filterStatus, setFilterStatus] = useState<TodoStatus>(TodoStatus.all);

  const filteredTodos = useMemo(() => {
    return todos.filter(todo => {
      switch (filterStatus) {
        case TodoStatus.active:
          return !todo.completed;
...

```

move all logic to context and return from context all things that are needed for components
with context your components shouldn't have props at all